aiida_pseudo/cli/install.py

```diff
@@ -23,43 +24,56 @@ def cmd_install():
 
 
 @cmd_install.command('family')
-@click.argument('archive', type=types.FileOrUrl(mode='rb'))
+@click.argument('archive_or_folder', type=types.PathOrUrl(exists=True, file_okay=True))
 @click.argument('label', type=click.STRING)
 @options_core.DESCRIPTION(help='Description for the family.')
 @options.ARCHIVE_FORMAT()
-@options.FAMILY_TYPE()
+@options.PSEUDOS_TYPE()
 @options.TRACEBACK()
 @decorators.with_dbenv()
-def cmd_install_family(archive, label, description, archive_format, family_type, traceback):  # pylint: disable=too-many-arguments
-    """Install a standard pseudo potential family from an ARCHIVE on the local file system or from a URL.
+def cmd_install_family(archive_or_folder, label, description, archive_format, pseudos_type, traceback):  # pylint: disable=too-many-arguments
+    """Install a standard pseudo potential family from a FOLDER or an ARCHIVE (on the local file system or from a URL).
 
-    The command will attempt to infer the archive format from the filename extension of the ARCHIVE. If this fails, the
-    archive format can be specified explicitly with the archive format option, which will also display which formats
-    are supported.
+    The command will attempt first to recognize the passed `archive_or_folder` as a folder in the local system. If not,
+    `archive_or_folder` is assumed to be an archive and the command will attempt to infer the archive format from the
+    filename extension of the ARCHIVE. If this fails, the archive format can be specified explicitly with the archive
+    format option, which will also display which formats are supported.
 
-    By default, the command will create a base `PseudoPotentialFamily`, but the type can be changed with the family
+    By default, the command will create a base `PseudoPotentialFamily`, but the type can be changed with the pseudos
     type option. If the base type is used, the pseudo potential files in the archive *have* to have filenames that
     strictly follow the format `ELEMENT.EXTENSION`, because otherwise the element cannot be determined automatically.
     """
     from .utils import attempt, create_family_from_archive
+    from aiida_pseudo.groups.family.pseudo import PseudoPotentialFamily
 
-    # The `archive` is now either a `http.client.HTTPResponse` or a normal filelike object, so we get the original file
-    # name in a different way.
-    try:
-        suffix = os.path.basename(archive.url)
-    except AttributeError:
-        suffix = os.path.basename(archive.name)
-
-    # Write the content of the archive to a temporary file, because `create_family_from_archive` does currently not
-    # accept filelike objects because the underlying `shutil.unpack_archive` does not. Likewise, `unpack_archive` will
-    # attempt to deduce the archive format from the filename extension, so it is important we maintain the original
-    # filename. Of course if this fails, users can specify the archive format explicitly wiht the corresponding option.
```

Comment:
Please reinstate this comment as it is not obvious why we need to preserve the suffix